### PR TITLE
feat(auth): add authentication landing page

### PR DESCRIPTION
**Pull request by laura-thresh**
----

Notes

----

## PR Checklist
- [x] Documentation has been checked by at least one team member
- [x] Links work to navigate documentation
- [x] Images stored in `_images`
- [x] Spell checked

> If some of the above are not completed why?
> **REASON HERE**

### CODE REVIEW

File: docs/authentication/scenarios.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+---
+
+## Applications authenticated with Pure Farming
+**Prerequisite-** Third-party businesses/Third-party application developers need to send an email to info@purefarming.com to get the client ID and client secret. 
+
+!(OAuth_Scenarios_ClientSecret.svg)
+
+The Pure Farming OAuth 2.0 endpoints supports any language or run-time that can make HTTP requests, for example Java, PHP, Python, and ASP.NET. 
+
+Pure Farming APIs use PKCE, short for proof key for code exchange, authorisation code flow. In order to make Pure Farming API requests, users will need to obtain an access token and refresh token. The application needs to redirect the user to the authorisation endpoint https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth with the client ID and code challenge.
```
Comment (danielR9) [inline docs/authentication/scenarios.md:27]:
```suggestion
Pure Farming APIs use proof key for code exchange (PKCE) authorisation code flow. In order to make Pure Farming API requests, users will need to obtain an access token and refresh token. The application needs to redirect the user to the authorisation endpoint https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth with the client ID and code challenge.  
```

File: docs/authentication/index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+taxonomy:
+    category:
+        - root
+    post_tag:
+        - root
+
+---
+
+# OAuth 2.0
+OAuth 2.0, short for Open Authorisation, is an authorisation protocol that enables applications to obtain limited access to resources on an HTTP service. This authorisation standard protects user data by providing access to the data without revealing the user’s identity or credentials. It uses username and password tokens instead.
```
Comment (danielR9) [inline docs/authentication/index.md:16]:
```suggestion
OAuth 2.0 is an authorisation protocol that enables applications to obtain limited access to resources on an HTTP service. This authorisation standard protects user data by providing access to the data without revealing the user’s identity or credentials. It uses username and password tokens instead.  
```

I'm just not a fan of "short for" - isn't OAuth2.0 standard enough we don't need to define it?

**Review by danielR9 — APPROVED** [review]
Couple of minor grammar things - feel free to ignore them. I know the text hasn't actually changed in this PR and this stuff is already on devhub

File: docs/authentication/basic-steps.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+---
+
+# Basic Steps
+All applications need to follow a basic pattern when accessing Pure Farming APIs using OAuth 2.0. Here are the steps: 
+- [**Step 1:** Obtain Client ID and Client Secret](#step1)
+- [**Step 2:** Redirect the User’s browser to the Authorisation Endpoint with the Client ID and Code Challenge](#step2)
+- [**Step 3:** Send Authorization Code and Code Verifier](#step3)
+- [**Step 4:** Send Access Token to get Data](#step4)
+- [**Step 5:** Refresh the access token](#step5)
```
Comment (daniel-buchanan) [inline docs/authentication/basic-steps.md:21]: These will all need to be changed, the way that the code in WP generates the links is as follows:
`Step 1: Obtain Client ID and Client Secret` becomes `#step-1:-obtain-client-id-and-client-secret`

File: docs/authentication/basic-steps.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+**Note:** When you email us, please mention in the email whether you want to use the refresh tokens. It is worth noting that this is an interim measure until such a time as we are able to provide self-service tools to developers. 
+
+---
+
+## Step 2: Redirect the User’s browser to the Authorisation Endpoint with the Client ID and Code Challenge
+Your application can only access private data using Pure Farming APIs if it has an access token that grants access to the data or endpoints provided by API. For obtaining the Access Token and Refresh Token, the application should redirect the user’s browser to the Authorization endpoint (https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth) with the Client ID and Code Challenge to get the Authorization code. If the user is logged in, they will be redirected to the application with the authorization code. If the user is not logged in, they would need to log in. Once they log in, they will be redirected to the application with the authorization code.  
+
+**Your app needs to send the Client ID and Code Challenge to the following URL:**
+
+https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/token
```
Comment (daniel-buchanan) [inline docs/authentication/basic-steps.md:37]:
```suggestion
```
https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/token 
```
```

File: docs/authentication/index.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+1. **Flexible protocol**– OAuth 2.0 is a very flexible protocol that relies on SSL/TLS (Secure Sockets Layer that ensures data between the web server and browsers remain private) to save user access tokens.  
+2. **Easy to implement and provides strong authentication**– In addition to being a flexible protocol, OAuth 2.0 is also easy to implement due to its well-documented nature and easy availability of supporting libraries.
+3. **Allows limited access to the user’s data**– OAuth 2.0 uses JWT’s to share data about the authenticated user – primarily their “subject” and any “scopes” that they have been granted access to. If a user has granted permission, some of the “profile” information may be included such as: email, name, etc.  
+4. **Usable in a variety of solutions**– The access to the resources is realized via HTTP/HTTPs with the token indicated in the headers. This allows OAuth usage in almost any solution: in mobile and desktop applications, on various web portals, and even in browser plug-ins. 
+
+---
+
+## Endpoints
+**Authorisation:** https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth
```
Comment (daniel-buchanan) [inline docs/authentication/index.md:43]:
```suggestion
**Authorisation:** `https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth`
```

File: docs/authentication/index.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+2. **Easy to implement and provides strong authentication**– In addition to being a flexible protocol, OAuth 2.0 is also easy to implement due to its well-documented nature and easy availability of supporting libraries.
+3. **Allows limited access to the user’s data**– OAuth 2.0 uses JWT’s to share data about the authenticated user – primarily their “subject” and any “scopes” that they have been granted access to. If a user has granted permission, some of the “profile” information may be included such as: email, name, etc.  
+4. **Usable in a variety of solutions**– The access to the resources is realized via HTTP/HTTPs with the token indicated in the headers. This allows OAuth usage in almost any solution: in mobile and desktop applications, on various web portals, and even in browser plug-ins. 
+
+---
+
+## Endpoints
+**Authorisation:** https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth 
+
+**Token:** https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/token
```
Comment (daniel-buchanan) [inline docs/authentication/index.md:45]:
```suggestion
**Token:** `https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/token`
```

File: docs/authentication/index.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+4. **Usable in a variety of solutions**– The access to the resources is realized via HTTP/HTTPs with the token indicated in the headers. This allows OAuth usage in almost any solution: in mobile and desktop applications, on various web portals, and even in browser plug-ins. 
+
+---
+
+## Endpoints
+**Authorisation:** https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/auth 
+
+**Token:** https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/token 
+
+**User Info:** https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/userinfo
```
Comment (daniel-buchanan) [inline docs/authentication/index.md:47]:
```suggestion
**User Info:** ~https://signin.purefarming.com/auth/realms/moa/protocol/openid-connect/userinfo`
```

File: docs/authentication/index.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+## Intended Audience for this Guide
+The following are the intended audience for this document:
+
+- **Non-Technical Users:** This guide provides an overview for people who want to use Pure Farming APIs and want to know how their developers could access the APIs using OAuth 2.0. 
+- **Application developers:** This guide is useful for application developers who want to integrate Pure Farming APIs into their applications. 
+
+---
+
```
Comment (daniel-buchanan) [inline docs/authentication/index.md:31]: We need a ToC or something that links to the other pages here.

File: docs/authentication/scenarios.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+taxonomy:
+    category:
+        - root
+    post_tag:
+        - root
+
+---
+# Scenarios
+- [Applications authenticated with Pure Farming](#authenticated)
+- [Applications not authenticated with Pure Farming](#NotAuthenticated)
```
Comment (daniel-buchanan) [inline docs/authentication/scenarios.md:16]: Same rules as above apply here.

**Review by daniel-buchanan — CHANGES_REQUESTED** [review]
See comments.

**Review by daniel-buchanan — DISMISSED** [review]

**Review by daniel-buchanan — APPROVED** [review]

**Pull request merged**